crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -90,6 +90,32 @@ def addons
       end
     end
 
+    def repocheck
+      # FIXME: once we start working on 7 to 8 upgrade we have to adapt the sles version
+      zypper_stream = Hash.from_xml(
+        `sudo /usr/bin/zypper-retry --xmlout products`
+      )["stream"]
+
+      {}.tap do |ret|
+        if zypper_stream["message"] =~ /^System management is locked/
+          errors.add(
+            :base,
+            I18n.t("api.crowbar.zypper_locked", zypper_locked_message: zypper_stream["message"])
+          )
+          return ret
+        end
+
+        products = zypper_stream["product_list"]["product"]
+
+        ret[:os] = {
+          available: repo_version_available?(products, "SLES", "12.2")
+        }
+        ret[:cloud] = {
+          available: repo_version_available?(products, "suse-openstack-cloud", "8")
+        }
```

Comment:
If the product name won't be translated into other languages, then it could provided by the API and consumed through the UI.
Otherwise, just a product key should be required. But keep in mind that whenever a new product is added in the backend, its name should also be created/updated on Angular's language file to match the new product key and let the UI know what should be displayed.

Let me know, @vuntz if we need to analyse it more to make a decision.